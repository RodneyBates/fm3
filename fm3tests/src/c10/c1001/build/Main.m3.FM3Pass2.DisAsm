 ItkEOF()
 ItkLeftEnd()
 ItkModuleLt(L1(16_1),[3,0])
 ItkDeclScopeLt(L1(16_1))
 ItkOpenScopeLt(L1(16_1))
 ItkOpenDeclListLt(L1(16_1),[5,2])
 ItkProcDefLt(Id2"P",[5,2])
 ItkDeclNo(D3,[5,12])
 ItkDeclScopeLt(L2(16_2))
 ItkSignatureProperLt([5,14])
 ItkFormalsListLt(L2(16_2),[5,16])
 ItkVALUEFormalLt([5,16])
 ItkVALUEFormalIdListLt(L1(16_1),[5,22])
 ItkDeclNo(D1,[5,22])
 ItkVALUEFormalIdListRt(L1(16_1),[5,22])
 ItkVALUEFormalType([5,25])
 ItkDefTopExprNo(L2(16_2),[5,27])
 ItkVALUEFormalValue([5,35])
 ItkDeclValAbsent([5,35])
 ItkVALUEFormalRt([5,16])
 ItkFormalsListSep(L1(16_1),[5,35])
 ItkVARFormalLt([5,37])
 ItkVARFormalIdListLt(L1(16_1),[5,41])
 ItkDeclNo(D2,[5,41])
 ItkVARFormalIdListRt(L1(16_1),[5,41])
 ItkVARFormalType([5,44])
 ItkDefTopExprNo(L1(16_1),[5,46])
 ItkVARFormalValue([5,54])
 ItkDeclValAbsent([5,54])
 ItkVARFormalRt([5,37])
 ItkFormalsListRt(L2(16_2),[5,16])
 ItkResultTypeAbsent([6,2])
 ItkRaisesSetLt(L0(16_0),[6,2])
 ItkRaisesSetRt(L0(16_0),[6,2])
 ItkSignatureProperRt([5,14])
 ItkProcDefBody([6,2])
 ItkDeclScopeRt(L2(16_2))
 ItkOpenScopeLt(L2(16_2))
 ItkDeclScopeLt(L3(16_3))
 ItkOpenScopeLt(L3(16_3))
 ItkOpenDeclListLt(L0(16_0),[6,4])
 ItkOpenDeclListRt(L0(16_0),[6,4])
 ItkStmtListLt(L0(16_0),[7,4])
 ItkStmtListRt(L0(16_0),[7,4])
 ItkOpenScopeRt(L3(16_3))
 ItkDeclScopeRt(L3(16_3))
 ItkOpenScopeRt(L2(16_2))
 ItkProcDefRt(Id2"P",[7,8])
 ItkOpenDeclListRt(L1(16_1),[5,2])
 ItkModuleBegin(L1(16_1),[9,2])
 ItkStmtListLt(L0(16_0),[10,2])
 ItkStmtListRt(L0(16_0),[10,2])
 ItkOpenScopeRt(L1(16_1))
 ItkDeclScopeRt(L1(16_1))
 ItkModuleRt(L1(16_1),[3,0])
 ItkRightEnd()
 ItkBOF()
     0
