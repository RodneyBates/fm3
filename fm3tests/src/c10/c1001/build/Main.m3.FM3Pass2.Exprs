From unit expression map, Expr No 0 NIL

From unit expression map, Expr No 1 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 1
 ExpPosition = [5,46]
 ExpOpcode = BOOLEAN
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 2 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 2
 ExpPosition = [5,27]
 ExpOpcode = INTEGER
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


