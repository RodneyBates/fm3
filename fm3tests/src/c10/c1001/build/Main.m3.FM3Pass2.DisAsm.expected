 ItkEOF()
 ItkLeftEnd()
 ItkModuleLt(L2(16_2),[3,0])
 ItkDeclScopeLt(L1(16_1))
 ItkOpenDeclListLt(L1(16_1),[5,2])
 ItkProcDefLt(Id3"P",[5,2])
 ItkDeclScopeLt(L2(16_2))
 ItkSignatureProperLt([5,14])
 ItkFormalsListLt(L2(16_2),[5,16])
 ItkVALUEFormalLt([5,16])
 ItkVALUEFormalIdListLt(L1(16_1),[5,22])
 ItkDeclNo(D3,[5,22])
 ItkVALUEFormalIdListRt(L1(16_1),[5,22])
 ItkVALUEFormalType([5,25])
 ItkDefTopExprNo(L2(16_2),[5,27])
 ItkVALUEFormalValue([5,35])
 ItkDeclValAbsent([5,16])
 ItkVALUEFormalRt([5,16])
 ItkFormalsListSep(L1(16_1),[5,35])
 ItkVARFormalLt([5,37])
 ItkVARFormalIdListLt(L1(16_1),[5,41])
 ItkDeclNo(D4,[5,41])
 ItkVARFormalIdListRt(L1(16_1),[5,41])
 ItkVARFormalType([5,44])
 ItkDefTopExprNo(L1(16_1),[5,46])
 ItkVARFormalValue([5,54])
 ItkDeclValAbsent([5,37])
 ItkVARFormalRt([5,37])
 ItkFormalsListRt(L2(16_2),[5,16])
 ItkResultTypeAbsent([6,2])
 ItkRaisesSetLt(L0(16_0),[6,2])
 ItkRaisesSetRt(L0(16_0),[6,2])
 ItkSignatureProperRt([5,14])
 ItkProcDefBody([6,2])
 ItkDeclScopeLt(L3(16_3))
 ItkOpenDeclListLt(L0(16_0),[6,4])
 ItkOpenDeclListRt(L0(16_0),[6,4])
 ItkStmtListLt(L0(16_0),[7,4])
 ItkStmtListRt(L0(16_0),[7,4])
 ItkOpenScopeRt(L3(16_3))
 ItkDeclScopeRt(L3(16_3))
 ItkOpenScopeRt(L2(16_2))
 ItkDeclScopeRt(L2(16_2))
 ItkProcDefRt(Id3"P",[5,2])
 ItkOpenDeclListRt(L1(16_1),[5,2])
 ItkModuleBegin(L2(16_2),[9,2])
 ItkStmtListLt(L0(16_0),[10,2])
 ItkStmtListRt(L0(16_0),[10,2])
 ItkOpenScopeRt(L1(16_1))
 ItkDeclScopeRt(L1(16_1))
 ItkModuleRt(L2(16_2),[3,0])
 ItkRightEnd()
 ItkBOF()
     0
